pkg/client/client_test.go

```diff
@@ -106,12 +172,22 @@ func TestClient(t *testing.T) {
 			if h.r.URL.String() != expURL {
 				t.Fatalf("should have %s for URL, got: %s", expURL, h.r.URL.String())
 			}
+
+			if _, _, ok := h.r.BasicAuth(); ok {
+				t.Fatalf("basic auth should not be set")
+			}
```

Comment:
I don't think this should be part of this test. We only need one test I think, verifying that client use given round tripper, which can as well be our test one.